draft-ietf-rats-corim.md

```diff
@@ -1066,6 +1072,328 @@ The following describes each member of the `concise-bom-tag` map.
+
+Note that what follows describes a simplified implementation.
+Verifiers claiming compliance with this specification must exhibit the same
+externally visible behavior as described here,
+they are not required to use the same internal data structures.
+For example, it is expected that the resources used during the initialisation
+phase can be amortised across multiple appraisals.
+
+## Appraisal Context initialisation
+
+The goal of the initialisation phase is to load the CoRIM Appraisal Context
+with objects such as CoRIM tags,
+cryptographic validation key material (e.g., raw public keys, root certificates,
+intermediate CA certificate chains), etc. that will be used in the subsequent
+Evidence Appraisal phase.
+
+### CoRIM Selection
+
+All available CoRIMs are collected.
+A Verifier may be pre-configured with a large number of tags describing many
+types of device.
+All CoRIMs are loaded at this stage, later stages will select the CoRIMs
+appropriate to the Evidence Appraisal step.
+
+CoRIMs that are not within their validity period, or that cannot be associated
+with an authenticated and authorised source MUST be discarded.
+
+CoRIM that are secured by a cryptographic mechanism such as a signature which
+does not pass validation MUST be discarded.
+
+Other selection criteria MAY be applied.
+
+For example, if the Evidence format is known in advance, tags that do not match
+the expected profile can be readily discarded.
+
+The selection process MUST yield at least one usable tag.
+
+### CoBOM Extraction
+
+All the available Concise Bill Of Material (CoBOMs) tags are then collected
+from the selected CoRIMs.
+
+The verifier MUST activate all tags referenced by a CoBOM.
+
+After the verifier has processed all CoBOMs it MUST discard any tags which have
+not been activated by a CoBOM.
+
+### Tags Identification and Validation
+
+The verifier chooses tags -- including Concise Module ID Tags (CoMID, {{sec-comid}}),
+Concise Software ID Tags (CoSWID, {{-coswid}}),
+and/or Concise Trust Anchor Stores (CoTS, {{?I-D.ietf-rats-concise-ta-stores}}) --
+from the selected CoRIMs.
+
+The verifier MUST discard all tags which are not syntactically and semantically
+valid.
+In particular, any cross-referenced triples (e.g., CoMID-CoSWID linking triples)
+MUST be successfully resolved.
+
+### Appraisal Context Construction
+
+All of the validated and potentially useful tags are loaded into the Appraisal Context. 
+Each tag is loaded into the Appraisal Context as a single unit, later stages
+will accept or ignore the whole tag.
+
+This concludes the initialisation phase.
+
+## Evidence Collection
+
+In the evidence collection phase the verifier communicates with attesters to
+collect evidence.
+
+The first part of the Evidence collection phase does not perform any
+cryptographic validation.
+This allows verifiers to use untrusted code for their initial Evidence collection.
+
+The results of the evidence collection are protocol specific data and transcripts
```

Comment:
Inconsistent use of terms: 'Evidence collection' or 'evidence collection' but not both.